PythonWrapperGenerator/TypeUtilities.cpp

```diff
@@ -0,0 +1,157 @@
+			translated = "::";
+			break;
+		default:
+			// In version 3.6 this is only NestedNameSpecifier::Super, which is a Microsoft specific extension (_super).
+			Q_ASSERT(false);
+			break;
+	}
+	return translated;
+}
+
+QString TypeUtilities::templateArgToString(const clang::TemplateArgument& templateArg)
+{
+	switch (templateArg.getKind())
+	{
+		case clang::TemplateArgument::ArgKind::Null:
+			return {};
+		case clang::TemplateArgument::ArgKind::Type:
+			return typePtrToString(templateArg.getAsType().getTypePtr());
+		case clang::TemplateArgument::ArgKind::Declaration:
+			return QString::fromStdString(templateArg.getAsDecl()->getQualifiedNameAsString());
+		case clang::TemplateArgument::ArgKind::NullPtr:
+			return {}; //new OOModel::NullLiteral();
+		case clang::TemplateArgument::ArgKind::Integral:
+			return {}; //new OOModel::IntegerLiteral(templateArg.getAsIntegral().getLimitedValue());
+//		case clang::TemplateArgument::ArgKind::Template:
+//			return new OOModel::ReferenceExpression(
+//						QString::fromStdString(templateArg.getAsTemplate().getAsTemplateDecl()->getNameAsString()));
+//		case clang::TemplateArgument::ArgKind::TemplateExpansion:
+//			// TODO: add support
+//			log_->writeError(className_, location, CppImportLogger::Reason::OTHER,
+//								  "Unsupported TemplateArgument EXPANSION");
+//			return createErrorExpression("Unsupported TemplateArgument EXPANSION");
+//		case clang::TemplateArgument::ArgKind::Expression:
+//			return exprVisitor_->translateExpression(templateArg.getAsExpr());
+//		case clang::TemplateArgument::ArgKind::Pack:
+//			// TODO: add support
+//			log_->writeError(className_, location, CppImportLogger::Reason::OTHER,
+//								  "Unsupported TemplateArgument PACK");
+//			return createErrorExpression("Unsupported TemplateArgument PACK");
+		default:
+			qDebug() << "Unsupported template arg" << templateArg.getKind();
+			return {};
+	}
+}
+
+QString TypeUtilities::typePtrToString(const clang::Type* type)
+{
+	if (auto recordType = llvm::dyn_cast<clang::RecordType>(type))
+	{
+		return QString::fromStdString(recordType->getDecl()->getQualifiedNameAsString());
+	}
+	else if (auto elaboratedType = llvm::dyn_cast<clang::ElaboratedType>(type))
+	{
+		// TODO: this might also have a keyword in front (like e.g. class, typename)
+		return typePtrToString(elaboratedType->getNamedType().getTypePtr());
+	}
+	else if (auto templateSpecialization = llvm::dyn_cast<clang::TemplateSpecializationType>(type))
+	{
+		QString baseName = QString::fromStdString(
+					templateSpecialization->getTemplateName().getAsTemplateDecl()->getQualifiedNameAsString());
+		if (baseName == "Super")
+		{
+			// We "unwrap" the Super baseclass
+			unsigned int numArgs = templateSpecialization->getNumArgs();
+			if (numArgs == 1)
+			{
+				auto argument = templateSpecialization->getArg(0);
+				return templateArgToString(argument);
+			}
+			else qDebug() << "Not 1 argument in super template";
```

Comment:
I see there are still quite a few uses of `qDebug`. Is the idea to remove them later as the code matures?